qemu/tests/timedrift_check_after_load_vm.py

```diff
@@ -0,0 +1,157 @@
+import re
+import time
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+from virttest import arch
+
+from qemu.tests.qemu_guest_agent import QemuGuestAgentTest
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Clock check after savevm and loadvm
+    Note: Run this test case, make sure that guest agent is installed
+    and start inside the guest.
+
+    1. On host, load kvm module with "kvmclock_periodic_sync=N"
+    2. On host, sync time with ntp server
+    3. Boot guest with qemu guest agent
+    4. On guest, sync time with ntp server
+    5. On guest, stop chronyd
+    6. Run samevm and loadvm
+    7. Run qemu-guest-agent command: guest-set-time
+    8. Run qmp command: rtc-reset-reinjection
+    9. On guest, query the time offset with ntp server,
+       it should be less than 3 seconds
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _load_kvm_module_with_kvmclock_periodic_sync(module_param):
+        """
+        Load kvm module with kvmclock_periodic_sync=N/Y
+
+       :params module_param: the value of kvmclock_periodic_sync
+       """
+        error_context.context("_load_kvm_module_with_kvmclock_periodic_sync=%s"
+                              % module_param, logging.info)
+        check_modules = arch.get_kvm_module_list()
+        error_context.context("check_module: '%s'" % check_modules, logging.info)
+        check_modules.reverse()
+        for module in check_modules:
+            rm_mod_cmd = "modprobe -r %s" % module
+            process.system(rm_mod_cmd, shell=True)
+        check_modules.reverse()
+        for module in check_modules:
+            load_mod_cmd = "modprobe %s" % module
+            if module == "kvm":
+                load_mod_cmd = "%s kvmclock_periodic_sync=%s" % (load_mod_cmd, module_param)
+            process.system(load_mod_cmd, shell=True)
+        check_mod_cmd = params["check_mod_cmd"]
+        if process.system_output(check_mod_cmd).decode() != module_param:
+            test.error("Cannot load kvm module with kvmclock_periodic_sync=%s"
+                       % module_param)
+
+    def setup():
+        """
+        On host, load kvm module with "kvmclock_periodic_sync=N"
+        sync time with ntp server and boot the guest
+        """
+        if arch.ARCH not in ('ppc64', 'ppc64le'):
+            _load_kvm_module_with_kvmclock_periodic_sync("N")
+        error_context.context("Sync host time with ntp server", logging.info)
+        ntp_cmd = params.get("ntp_cmd")
+        status = process.system(ntp_cmd, shell=True)
+        if status:
```

Comment:
To my understanding, status value can hardly be zero.  If have time drift, then it will execute test.cancel()?